src/dune_engine/rule_cache.ml

```diff
@@ -331,6 +331,20 @@ module Shared = struct
       | Already_present targets_and_digests ->
         Log.info [ Pp.textf "cache store skipped [%s]: already present" hex ];
         update_cached_digests ~targets_and_digests
+      | Error (Unix.Unix_error (Unix.EXDEV, "link", file)) ->
```

Comment:
It may be cleaner to catch this error in `Dune_cache.Local.store_artifacts` and return it in the `Error` variant below, so we have only one error path (we already do this with some other exceptions).

Reply:
Oh, sorry, never mind. It already comes from `Error`!